cmd/gaia/testnets/README.md

```diff
@@ -88,194 +99,200 @@ Your full node has been cleanly upgraded!
 
 Copy the testnet's `genesis.json` file and place it in `gaiad`'s config directory.
 
-```
+```bash
 mkdir -p $HOME/.gaiad/config
 cp -a $GOPATH/src/github.com/cosmos/cosmos-sdk/cmd/gaia/testnets/gaia-6002/genesis.json $HOME/.gaiad/config/genesis.json
 ```
 
 ### Add Seed Nodes
 
-Your node needs to know how to find peers. You'll need to add healthy seed nodes to `$HOME/.gaiad/config/config.toml`. Here are some seed nodes you can use: 
+Your node needs to know how to find peers. You'll need to add healthy seed nodes to `$HOME/.gaiad/config/config.toml`. Here are some seed nodes you can use:
 
-```
+```toml
 # Comma separated list of seed nodes to connect to
 seeds = "38aa9bec3998f12ae9088b21a2d910d19d565c27@gaia-6002.coinculture.net:46656,80a35a46ce09cfb31ee220c8141a25e73e0b239b@seed.cosmos.cryptium.ch:46656,80a35a46ce09cfb31ee220c8141a25e73e0b239b@35.198.166.171:46656,032fa56301de335d835057fb6ad9f7ce2242a66d@165.227.236.213:46656"
 ```
 
-If those seeds aren't working, you can find more seeds and persistent peers on the [Cosmos Explorer](https://explorecosmos.network/nodes). For more information on seeds and peers, [read this](https://github.com/tendermint/tendermint/blob/develop/docs/using-tendermint.md#peers).
+If those seeds aren't working, you can find more seeds and persistent peers on the [Cosmos Explorer](https://explorecosmos.network/nodes). Open the the `Full Nodes` pane and select nodes that do not have private (`10.x.x.x`) or [local IP addresses](https://en.wikipedia.org/wiki/Private_network). The `Persistent Peer` field contains the connection string. For best results use 4-6.
+
+For more information on seeds and peers, [read this](https://github.com/tendermint/tendermint/blob/develop/docs/using-tendermint.md#peers).
 
 ## Run a Full Node
 
 Start the full node with this command:
 
-```
+```bash
 gaiad start
 ```
 
 Check that everything is running smoothly:
 
-```
+```bash
 gaiacli status
 ```
 
 View the status of the network with the [Cosmos Explorer](https://explorecosmos.network). Once your full node syncs up to the current block height, you should see it appear on the [list of full nodes](https://explorecosmos.network/validators). If it doesn't show up, that's ok--the Explorer does not connect to every node.
 
-## Generate Keys
+## Generating Keys
+
+### A Note on Keys in Cosmos:
+
+There are three types of key representations that are used in this tutorial:
+
+- `cosmosaccaddr`
+  * Derived from account keys generated by `gaiacli keys add`
+  * Used to receive funds
+  * e.g. `cosmosaccaddr15h6vd5f0wqps26zjlwrc6chah08ryu4hzzdwhc`
+
+- `cosmosaccpub`
+  * Derived from account keys generated by `gaiacli keys add`
+  * e.g. `cosmosaccpub1zcjduc3q7fu03jnlu2xpl75s2nkt7krm6grh4cc5aqth73v0zwmea25wj2hsqhlqzm`
+
+- `cosmosvalpub`
+  * Generated when the node is created with `gaiad init`. Needed when staking.
```

Comment:
needed when staking? i thought this was needed to become a validator...?

Reply:
So my understanding is that this key is derived from the `~/.gaiad/config/priv_validator.json` when the node is created. You need to _use_ that address in the command when you stake. I'll make that description better.